packages/mask/src/plugins/Referral/types.ts

```diff
@@ -0,0 +1,303 @@
+    rewardToken?: FungibleTokenDetailed
+    referredToken?: FungibleTokenDetailed
+}
+
+export interface DepositDialogInterface {
+    deposit?: DepositProps
+}
+
+type TransactionProps =
+    | {
+          status: TransactionStatus.CONFIRMATION
+          title: string
+          subtitle?: string
+      }
+    | {
+          status: TransactionStatus.CONFIRMED
+          actionButton: {
+              label: string
+              onClick: (token?: FungibleTokenDetailed) => void
+          }
+          transactionHash: string
+      }
+    | {
+          status: TransactionStatus.FAILED
+          actionButton: {
+              label: string
+              onClick: () => void
+          }
+          subtitle?: string
+      }
+
+export interface TransactionDialogInterface {
+    onClose?: () => void
+    transaction?: TransactionProps
+}
+
+export interface DialogInterface {
+    hideBackBtn?: boolean
+    hideAttrLogo?: boolean
+    adjustFarmDialog?: AdjustFarmRewardsInterface
+    depositDialog?: DepositDialogInterface
+    transactionDialog?: TransactionDialogInterface
+}
+
+export interface PageInterface {
+    pageType?: PagesType
+    onClose?: () => void
+    continue: (currentPage: PagesType, nextPage: PagesType, title?: string, props?: DialogInterface) => void
+    onChangePage?: (page: PagesType, title?: string, props?: DialogInterface) => void
+}
+
+export interface ChainAddressProps {
+    chainId: number
+    address: EvmAddress
+    isNative: boolean
+}
+
+export enum TransactionStatus {
+    CONFIRMATION = 'Confirmation',
+    CONFIRMED = 'CONFIRMED',
+    FAILED = 'FAILED',
+}
+
+// effects
+export type VerifierEffect = { nonce: number; proof: string[] }
+export type FarmRewardStruct = {
+    farmHash: FarmHash
+    value: {
+        type: 'string'
+        hex: string
+    }
+}
+export type HarvestRequest = {
+    rewardTokenDefn: ChainAddress
+    rewards: FarmRewardStruct[]
+}
+
+export enum RpcMethod {
+    oracle_chainId = 'oracle_chainId',
+    oracle_getDerivedBlockByHash = 'oracle_getDerivedBlockByHash',
+    oracle_getBundleReceipt = 'oracle_getBundleReceipt',
+    oracle_getDerivedBlockByNumber = 'oracle_getDerivedBlockByNumber',
+    oracle_getOperationalAddress = 'oracle_getOperationalAddress',
+    oracle_getTimePromise = 'oracle_getTimePromise',
+    oracle_sendProofOfRecommendationOrigin = 'oracle_sendProofOfRecommendationOrigin',
+    oracle_sendProofOfRecommendation = 'oracle_sendProofOfRecommendation',
+}
+export type JsonRpcRequestId = number | string
+export interface JsonRpcResponse {
+    jsonrpc: '2.0'
+    // Has a value for successful requests, might have a value for failed requests
+    id: JsonRpcRequestId | null
+    // If result has a value, it indicates success
+    result?: any
+    // If error has a value, indicates failure
+    error?: {
+        code: number // Json Rpc compatible code
+        message: string
+        data?: any
+    }
+}
+
+// entitlements
+export interface Entitlement {
```

Comment:
FYI: We have a package, `web3-contracts`, which generates a class from ABI definitions.

Reply:
Thanks, added ABI for another contract and use this approach, didn’t add it for this Entitlement event, because it's only one event.